plugins/vlc-video/vlc-video-source.c

```diff
@@ -394,12 +394,97 @@ static int vlcs_audio_setup(void **p_data, char *format, unsigned *rate,
 	return 0;
 }
 
+static inline void vlc_split_media_path(const struct dstr *new_path,
+	struct dstr *path_only, struct dstr *options)
+{
+	int i;
+	const char* path = new_path->array;
+	bool quoting = false;
+	int options_at = 0;
+	int path_len;
+
+	for (i = 0; path[i]; ++i) {
+		if (path[i] == '"') {
+			quoting = !quoting;
+		}
+		if (!quoting && path[i+1]==' ' && path[i+2]==':') {
```

Comment:
Style: Place spaces around the equality operator (==).  There is some inconsistency in the existing codebase about whether or not there should be spaces around the plus signs (+) inside the square brackets for the array index indicators, so I'll let @jp9000 comment on that.